plugins/org.eclipse.emf.ecore.xmi/src/org/eclipse/emf/ecore/xmi/impl/XMLHandler.java

```diff
@@ -804,6 +810,11 @@ public void notationDecl(String name, String publicId, String systemId) throws S
   @Override
   public InputSource resolveEntity(String publicId, String systemId) throws SAXException
   {
+    if (!resolveEntities)
+    {
+      return null;
```

Comment:
```suggestion
		return new InputSource(new StringReader("")); //$NON-NLS-1$
```

```
     * @return An InputSource object describing the new input source,
     *         or null to request that the parser open a regular
     *         URI connection to the system identifier.

```
It would be even better to also adapt org.eclipse.emf.ecore.xmi.impl.XMLParserPoolImpl.makeParser(Map<String, Boolean>, Map<String, ?>)
and org.eclipse.emf.ecore.xmi.impl.XMLLoadImpl.makeParser()
to use
```
		// prohibit the use of all protocols by external entities:
		parser.setProperty(XMLConstants.ACCESS_EXTERNAL_DTD, ""); //$NON-NLS-1$
		parser.setProperty(XMLConstants.ACCESS_EXTERNAL_SCHEMA, ""); //$NON-NLS-1$
```

Reply:
My goal is to provide options, not to cripple the framework for absolutely everyone.  In any case, that is only optionally used when one uses a pooling option.